From bb9af3d5c7a6d44789bab58b650840983cfafe9c Mon Sep 17 00:00:00 2001
From: Michael Auchter <michael.auchter@ni.com>
Date: Fri, 4 Oct 2019 12:58:48 -0500
Subject: [PATCH 08/37] ni: zynqmp: load eeproms at init

Load contents from all attached EEPROMs at init, for later use during
board initialization.

Signed-off-by: Michael Auchter <michael.auchter@ni.com>
---
 board/ni/zynqmp/board.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/board/ni/zynqmp/board.c b/board/ni/zynqmp/board.c
index e9583dd0fb1..0161840b25c 100644
--- a/board/ni/zynqmp/board.c
+++ b/board/ni/zynqmp/board.c
@@ -28,6 +28,10 @@ struct eeprom_ctx {
 };
 
 static struct eeprom_ctx mb_eeprom;
+static struct eeprom_ctx db0_eeprom;
+static struct eeprom_ctx db1_eeprom;
+static struct eeprom_ctx dioaux_eeprom;
+static struct eeprom_ctx clkaux_eeprom;
 
 uint32_t tlv_eeprom_crc(const struct tlv_eeprom *eeprom)
 {
@@ -85,7 +89,6 @@ int zynq_board_read_rom_ethaddr(unsigned char *ethaddr)
 {
 	const struct usrp_eeprom_mac_addr *hwaddr;
 
-	load_eeprom("ni,mb-eeprom", &mb_eeprom);
 	if (!mb_eeprom.valid)
 		return -1;
 
@@ -237,5 +240,11 @@ int board_late_init(void)
 		break;
 	}
 
+	load_eeprom("ni,mb-eeprom", &mb_eeprom);
+	load_eeprom("ni,db0-eeprom", &db0_eeprom);
+	load_eeprom("ni,db1-eeprom", &db1_eeprom);
+	load_eeprom("ni,dioaux-eeprom", &dioaux_eeprom);
+	load_eeprom("ni,clkaux-eeprom", &clkaux_eeprom);
+
 	return 0;
 }
-- 
2.43.0

